platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -1378,11 +1384,11 @@ std::exception_ptr OfflineDatabase::resetDatabase() try {
     return std::current_exception();
 }
 
-void OfflineDatabase::reopenDatabaseReadOnlyForTesting() {
-    readOnly = true;
-
+void OfflineDatabase::reopenDatabaseReadOnly(bool readOnly_) {
+    if (readOnly == readOnly_) return;
     try {
         cleanup();
+        readOnly = readOnly_;
         initialize();
     } catch (...) {
         handleError("reopen database read-only");
```

Comment:
I think `readOnly` flag should be set to original value, if initialization failed.

Reply:
It wouldn't make sense imo, as the previous db has been already deleted - no need to restore settings for it. In order to continue working `initialize()` must succeed anyway..

Reply:
I would switch it back to an original value, or even go one step further. If mode change failed, set flag to old one and initialize again. If rolling back failed, throw exception.

Reply:
I'm afraid it won't be obvious for the client, it'd be impossible to watch the db storage state